squeezenet.py

```diff
@@ -0,0 +1,172 @@
+from keras.layers import Input, merge
+from keras.layers.convolutional import Convolution2D, MaxPooling2D
+from keras.layers.core import Dropout, Activation
+from keras.layers.pooling import GlobalAveragePooling2D
+from keras.models import Model
+from keras import backend as K
+from keras.utils.layer_utils import convert_all_kernels_in_model
+from keras.utils.data_utils import get_file
+from keras.preprocessing import image
+from imagenet_utils import decode_predictions, preprocess_input
+import numpy as np
+import warnings
+
+'''SqueezeNet model for Keras.
+
+# Reference:
+- [SqueezeNet: AlexNet-level accuracy with 50x fewer parameters and <0.5MB model size](https://arxiv.org/abs/1602.07360)
+
+# Keras Project Reference:
+
+- [keras-squeezenet](https://github.com/rcmalli/keras-squeezenet)
+
+# Original Project Reference:
+
+- [Original Squeezenet](https://github.com/DeepScale/SqueezeNet)
+
+'''
+
+TH_WEIGHTS_PATH = 'PATH/squeezenet_weights_th_dim_ordering_th_kernels.h5'
+TF_WEIGHTS_PATH = 'PATH/squeezenet_weights_tf_dim_ordering_tf_kernels.h5'
+
+
+# Modular function for Fire Node
+
+def fire_module(x, fire_id, squeeze=16, expand=64):
+    sq1x1, exp1x1, exp3x3, relu = "squeeze1x1", "expand1x1", "expand3x3", "relu_"
+    s_id = 'fire' + str(fire_id) + '/'
+
+    if K.image_dim_ordering() == 'tf':
+        c_axis = 3
+    else:
+        c_axis = 1
+
+    x = Convolution2D(squeeze, 1, 1, border_mode='valid', name=s_id + sq1x1)(x)
+    x = Activation('relu', name=s_id + relu + sq1x1)(x)
+
+    left = Convolution2D(expand, 1, 1, border_mode='valid', name=s_id + exp1x1)(x)
+    left = Activation('relu', name=s_id + relu + exp1x1)(left)
+
+    right = Convolution2D(expand, 3, 3, border_mode='same', name=s_id + exp3x3)(x)
+    right = Activation('relu', name=s_id + relu + exp3x3)(right)
+
+    x = merge([left, right], mode='concat', concat_axis=c_axis, name=s_id + 'concat')
+    return x
+
+
+def SqueezeNet(nb_classes, finetune=False, input_tensor=None, weights=None):
```

Comment:
In general: please match the same constructor arguments as every other image classification model in the repo / in `keras.applications`: `(include_top=True, weights='imagenet', input_tensor=None)`